tests/integration/generate_fixtures/common.py

```diff
@@ -223,28 +227,19 @@ def mine_block(w3):
 
 
 def mine_transaction_hash(w3, txn_hash):
-    start_time = time.time()
-    w3.geth.miner.start(1)
-    while time.time() < start_time + 120:
-        try:
-            receipt = w3.eth.get_transaction_receipt(txn_hash)
-        except TransactionNotFound:
-            continue
-        if receipt is not None:
-            w3.geth.miner.stop()
-            return receipt
-        else:
-            time.sleep(0.1)
-    else:
-        raise ValueError(
-            "Math contract deploy transaction not mined during wait period"
-        )
+    w3.geth.miner.start()
+    try:
+        return w3.eth.wait_for_transaction_receipt(txn_hash, timeout=120)
+    finally:
+        w3.geth.miner.stop()
 
 
 def deploy_contract(w3, name, factory):
     name = name.upper()
     w3.geth.personal.unlock_account(w3.eth.coinbase, KEYFILE_PW)
-    deploy_txn_hash = factory.constructor().transact({"from": w3.eth.coinbase})
+    deploy_txn_hash = factory.constructor().transact(
+        {"from": w3.eth.coinbase, "gas": 5000000}
```

Comment:
The docs specify that transactions cost zero gas in `--dev` mode. Unclear why the gas is required to deploy the contract but still digging.

Reply:
Finding that my fixture doesnt generate with this gas value anymore. It might not be necessary with the recent configuration changes and mining.